src/client/mod.rs

```diff
@@ -75,6 +77,11 @@ impl<C: NetworkConnector> Client<C> {
         self.redirect_policy = policy;
     }
 
+    /// Set the UserAgent. Pass `None` to disable.
+    pub fn set_user_agent(&mut self, agent: Option<String>) {
```

Comment:
Yes, but then you have to add it for every request. If we provide a way to set default headers for every request, then you can cut all that boilerplate.